src/ngx_http_lua_socket_tcp.h

```diff
@@ -89,6 +89,7 @@ struct ngx_http_lua_socket_tcp_upstream_s {
     ngx_http_lua_co_ctx_t           *write_co_ctx;
 
     ngx_uint_t                       reused;
+    ngx_int_t                        tag_data; /* store tag data */
```

Comment:
Just like the `ctxt_ref`?  https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_ctx.c#L67